File: app/src/main/java/com/wolcano/musicplayer/music/ui/fragment/base/BaseFragmentInject.java
package com.wolcano.musicplayer.music.ui.fragment.base;

import android.os.Build;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.hwangjr.rxbus.RxBus;
import com.kabouzeid.appthemehelper.util.ColorUtil;
import com.wolcano.musicplayer.music.App;
import com.wolcano.musicplayer.music.content.Binder;
import com.wolcano.musicplayer.music.di.component.ApplicationComponent;
import com.wolcano.musicplayer.music.ui.activity.MainActivity;
import com.wolcano.musicplayer.music.utils.PermissionUtils;
import com.wolcano.musicplayer.music.widgets.StatusBarView;

public abstract class BaseFragmentInject extends Fragment {

    protected Handler handler;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        handler = new Handler(Looper.getMainLooper());
        Binder.bindIt(this, getView());
        RxBus.get().register(this);

    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setupComponent(((App) getActivity().getApplication()).getApplicationComponent());
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        PermissionUtils.onRequestPermissionsResult(requestCode, permissions, grantResults);
    }

    @Override
    public void onDestroy() {
        RxBus.get().unregister(this);
        super.onDestroy();
    }

    public void setStatusbarColor(int color, StatusBarView statusBarView) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            if (statusBarView != null) {
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                    statusBarView.setBackgroundColor(ColorUtil.darkenColor(color));
                    ((MainActivity) getActivity()).setLightStatusbarAuto(color);
                } else {
                    statusBarView.setBackgroundColor(color);
                }
            }
        }
    }

    public void setStatusbarColorAuto(StatusBarView statusBarView, int color) {
        setStatusbarColor(color, statusBarView);
    }

    public abstract void setupComponent(ApplicationComponent applicationComponent);
}

File: app/src/main/java/com/wolcano/musicplayer/music/ui/fragment/base/BaseFragmentInject.kt
package com.wolcano.musicplayer.music.ui.fragment.base

import android.os.Build
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.fragment.app.Fragment
import com.hwangjr.rxbus.RxBus
import com.wolcano.musicplayer.music.App
import com.wolcano.musicplayer.music.content.Binder.bindIt
import com.wolcano.musicplayer.music.di.component.ApplicationComponent
import com.wolcano.musicplayer.music.ui.activity.MainActivity
import com.wolcano.musicplayer.music.utils.ColorUtils
import com.wolcano.musicplayer.music.utils.PermissionUtils
import com.wolcano.musicplayer.music.widgets.StatusBarView

abstract class BaseFragmentInject : Fragment() {

    protected var handler: Handler? = null

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        handler = Handler(Looper.getMainLooper())
        bindIt(this, view)
        RxBus.get().register(this)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setupComponent((activity!!.application as App).getApplicationComponent())
    }

    override fun onRequestPermissionsResult(
        requestCode: Int,
        permissions: Array<String?>,
        grantResults: IntArray
    ) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
        PermissionUtils.onRequestPermissionsResult(requestCode, permissions, grantResults)
    }

    override fun onDestroy() {
        RxBus.get().unregister(this)
        super.onDestroy()
    }

    open fun setStatusbarColor(color: Int, statusBarView: StatusBarView?) {
        if (statusBarView != null) {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
                statusBarView.setBackgroundColor(ColorUtils.darkenColor(color))
                (activity as MainActivity?)?.setLightStatusbarAuto(color)
            } else {
                statusBarView.setBackgroundColor(color)
            }
        }
    }

    abstract fun setupComponent(applicationComponent: ApplicationComponent?)

}